File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        
        for (int i = 0; i < n; i++) {
            String line = Integer.toBinaryString(arr1[i] | arr2[i]);
            line = line.replace("0", " ");
            line = line.replace("1", "#");
            if (line.length() < n)
                line = " ".repeat(n - line.length()) + line;
            answer[i] = line;
        }
        
        return answer;
    }
}

File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    for i in range(n):
        line = bin(arr1[i] | arr2[i])[2:]
        line = line.replace('0', ' ')
        line = line.replace('1', '#')
        if len(line) < n:
            line = ' ' * (n - len(line)) + line
        answer.append(line)
    return answer